snapcraft/application.py

```diff
@@ -373,6 +379,16 @@ def create_app() -> Snapcraft:
 
 def main() -> int:
     """Run craft-application based snapcraft with classic fallback."""
+    if os.getenv("SNAPCRAFT_BUILD_ENVIRONMENT") == "managed-host":
+        snapcraft.ProjectOptions = snapcraft_legacy.ProjectOptions  # type: ignore
+        legacy.legacy_run()
```

Comment:
@mattculler we need to cover these lines with an unit test. Here's how I would do it:
- Create a test that mocks `legacy.legacy_run()` and `create_app()`
- Set `SNAPCRAFT_BUILD_ENVIRONMENT` to `managed-host` 
- call this `main()` function
- check that the `legacy_run()` mock *was* called, and the `create_app()` mock was *not* called

Reply:
Will do, appreciate the suggestion

Reply:
Hey @tigarmo so looking at the code, why would you expect `create_app` to not be called?  The only way I can see is if `legacy_run` is expected to call `sys.exit` or raise an exception.  If that's the case, then testing for the lack of a `create_app` call won't be useful, since `legacy_run` is mocked.

Reply:
oh you're right
in that case, you can add a `return` after that `legacy.legacy_run()` call to emphasize the fact that the execution doesn't progress beyond this

Reply:
Ok makes sense, thanks!

Reply:
https://github.com/canonical/snapcraft/pull/4642/commits/8750f211f84f45a2b0818252206a79d63623fd32